_data/gsoc-ideas.yml

```
@@ -30,13 +30,77 @@
   categories: [operating systems, programming languages]
 
-- title: "Streaming implementation of an XQuery subset"
-  description: "XQuery allows for transforming XML data in a declarative way. Based on the generic streaming query engine added into the [fs2-data](https://fs2-data.gnieh.org) library, write a frontend to implement a streamable subset of XQuery."
-  prereqs: Scala, ideally some experience with XQuery
-  difficulty: Hard. 
+- title: Web Components for Calico
+  description:
+    Calico is a reactive UI library built with Cats Effect and FS2. Web Components are a standard for creating framework-agnostic, reusable UI elements. The goal of this project is to enable Calico users to access the vast array of web components available by improving its DSL and code-generation.
+  prereqs: Scala, ideally experience with Web APIs
+  difficulty: Medium.
+  length: long
+  mentors: [armanbilge]
+  categories: [web, programming languages]
+
+- title: Upgrade sbt-typelevel to sbt 2
+  description:
+    sbt-typelevel is a plugin for sbt, the Scala build tool, used by hundreds of open source and enterprise projects. sbt 2 is in the final stages of development. We want to upgrade sbt-typelevel to sbt 2 and adopt its new features, such as "project matrix" for cross-building.
+  prereqs: Scala
+  difficulty: Medium.
+  length: long
+  mentors: [mzuehlke, armanbilge]
+  categories: [development tools]
+
+- title: Refresh Davenverse projects
+  description:
+    The Davenverse is a collection of several popular Typelevel libraries, including Mules and cats-scalacheck. Unfortunately, we have fallen behind on their maintenance. We want to move these libraries under the Typelevel org, refresh their build tooling, and bring them up-to-date to ensure their longevity.
+  prereqs: Scala
+  difficulty: Medium.
+  length: medium
+  mentors: [samspills, valencik]
+  categories: [development tools, programming languages]
+
+- title: Cats Effect & FS2 on WASM/WASI
+  description:
+    Web Assembly and its System Interface are emerging technologies for deploying secure, modular applications. The goal of this project is to prototype porting the Cats Effect runtime and FS2 streaming I/O to the Wasm/WASI platform, also possibly generating feedback for the Scala WASM and WASI teams.
+  prereqs: Scala, ideally some experience with Wasm/WASI
+  difficulty: Hard. Wasm/WASI support in Scala is experimental.
+  length: long
+  mentors: [armanbilge, tanishiking, valencik]
+  categories: [web, cloud, operating systems, programming languages]
+
+- title: Extensible log4cats interface
+  description:
+    log4cats is the de facto logging library for the Typelevel stack. Recently, a new API was proposed that overcomes current limitations of log4cats. The goal of this project is to adopt the new API in log4cats, migrate existing integrations to the new API, and create a compatibility layer with the old API.
+  prereqs: Scala
+  difficulty: Medium
```

sorry just some inconsistent punctuation that I noticed after the fact
```suggestion
  difficulty: Medium.
```